homeassistant/components/mqtt/__init__.py

```diff
@@ -147,23 +138,23 @@
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.All(
-            cv.deprecated(CONF_BIRTH_MESSAGE),  # Deprecated in HA Core 2022.3
-            cv.deprecated(CONF_BROKER),  # Deprecated in HA Core 2022.3
-            cv.deprecated(CONF_CERTIFICATE),  # Deprecated in HA Core 2022.11
-            cv.deprecated(CONF_CLIENT_ID),  # Deprecated in HA Core 2022.11
-            cv.deprecated(CONF_CLIENT_CERT),  # Deprecated in HA Core 2022.11
-            cv.deprecated(CONF_CLIENT_KEY),  # Deprecated in HA Core 2022.11
-            cv.deprecated(CONF_DISCOVERY),  # Deprecated in HA Core 2022.3
-            cv.deprecated(CONF_DISCOVERY_PREFIX),  # Deprecated in HA Core 2022.11
-            cv.deprecated(CONF_KEEPALIVE),  # Deprecated in HA Core 2022.11
-            cv.deprecated(CONF_PASSWORD),  # Deprecated in HA Core 2022.3
-            cv.deprecated(CONF_PORT),  # Deprecated in HA Core 2022.3
-            cv.deprecated(CONF_PROTOCOL),  # Deprecated in HA Core 2022.11
-            cv.deprecated(CONF_TLS_INSECURE),  # Deprecated in HA Core 2022.11
-            cv.deprecated(CONF_TLS_VERSION),  # Deprecated June 2020
-            cv.deprecated(CONF_USERNAME),  # Deprecated in HA Core 2022.3
-            cv.deprecated(CONF_WILL_MESSAGE),  # Deprecated in HA Core 2022.3
-            CONFIG_SCHEMA_BASE,
+            cv.removed(CONF_BIRTH_MESSAGE),  # Removed in HA Core 2023.4
+            cv.removed(CONF_BROKER),  # Removed in HA Core 2023.4
+            cv.removed(CONF_CERTIFICATE),  # Removed in HA Core 2023.4
+            cv.removed(CONF_CLIENT_ID),  # Removed in HA Core 2023.4
+            cv.removed(CONF_CLIENT_CERT),  # Removed in HA Core 2023.4
+            cv.removed(CONF_CLIENT_KEY),  # Removed in HA Core 2023.4
+            cv.removed(CONF_DISCOVERY),  # Removed in HA Core 2022.3
+            cv.removed(CONF_DISCOVERY_PREFIX),  # Removed in HA Core 2023.4
+            cv.removed(CONF_KEEPALIVE),  # Removed in HA Core 2023.4
+            cv.removed(CONF_PASSWORD),  # Removed in HA Core 2023.4
+            cv.removed(CONF_PORT),  # Removed in HA Core 2023.4
+            cv.removed(CONF_PROTOCOL),  # Removed in HA Core 2023.4
+            cv.removed(CONF_TLS_INSECURE),  # Removed in HA Core 2023.4
+            cv.removed(CONF_TLS_VERSION),  # Removed in HA Core 2023.4
+            cv.removed(CONF_USERNAME),  # Removed in HA Core 2023.4
+            cv.removed(CONF_WILL_MESSAGE),  # Removed in HA Core 2023.4
+            PLATFORM_CONFIG_SCHEMA_BASE,
```

Comment:
There is no config base left to be setup through `configuration.yaml`, only manual MQTT items can still be set up through yaml using the modern schema.